src/display/canvas.js

```diff
@@ -934,6 +934,12 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
             break;
           case 'BM':
             this.ctx.globalCompositeOperation = value;
+            // We only manually blend if the mode is not 'source-over'
+            // which is the only supported blend mode does no blending
+            // See issue https://github.com/mozilla/pdf.js/issues/5264
+            this.blendModeFallback =
+              this.ctx.globalCompositeOperation !== value &&
```

Comment:
If the mode is unsupported, `ctx. globalCompositeOperation` won't update with that value. This allows us to avoid a browser specific check. If this is considered too brittle, I can move this check to compatibility and use the method described in the issue.

Reply:
> This allows us to avoid a browser specific check.

That's sort of beside the point here, at least in my opinion. Browser detection is usually used *only* as a last resort in `src/shared/compatibility.js`, in rare cases where feature detection won't work correctly.

This PR would still mean adding compatibility hacks to the regular code-base, which is something we always want to avoid for a couple of reasons.
First of all, it adds overhead (even though it's probably small in this case) to all modern browsers, even those that have proper blend mode support.
Secondly, these sort of hacks increase the maintenance burden of the code in question, which we generally want to avoid.

Reply:
> This allows us to avoid a browser specific check.

That's sort of beside the point here, at least in my opinion. Browser detection is usually used *only* as a last resort in `src/shared/compatibility.js`, in rare cases where feature detection won't work correctly.

This PR would still mean adding compatibility hacks to the regular code-base, which is something we always want to avoid for a couple of reasons.
First of all, it adds a certain amount of overhead to all modern browsers, even those that have proper blend mode support.
Secondly, these sort of hacks increase the maintenance burden of the code in question, which we generally want to avoid.

Reply:
hey @Snuffleupagus, thanks for the feedback.

My intention was that in many cases where blend modes are used, even viewing critical content is not possible, (see example 2). It is definitely not correct behavior, but I believe that it will make the document more readable in many cases, at the expense of a neutral change in other cases. I think browser overhead is minimal here, as I've only added one `if` check and an operation (setting global alpha) that is done in many other places.

If you're not comfortable with this kind of fallback, would writing a polyfill for blend modes on unsupported browsers be feasible? To me, it sounds like even more overhead. I would still need to do feature/browser detection, and I would be adding a lot more math based and risky code to the code base. My thought process for that kind of a fix would be:

1. Select the area of whatever asset/shape/image we are about to add on our existing canvas
2. Calculate the new color for that area of the canvas using the formula for that mode
3. Render the new asset and apply the color to the area.

Reply:
> I think browser overhead is minimal here, as I've only added one if check and an operation (setting global alpha) that is done in many other places.

It's not simply about run-time overhead, it's in equal measure relevant to the mental overhead these hacks will add to the main code-base.

---

Please note that you might want to wait for feedback from of others contributors here, since I suppose opinions might vary as to if/how we want to address this.

However, even if the consensus would be that we're OK with the current hack, I think that there's a couple of changes that would be required here:
 - First of all, the added code needs to be enclosed in preprocessor tags to prevent it from being bundled into anything except `GENERIC` builds; i.e.
```javascript
if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
  // The new code goes here...
}
```
 - Secondly, you'd need to *very clearly* state in the comments that this is a polyfill for IE11 (in addition to the issue link).
 - Thirdly, do the cheapest comparison first, i.e. `this.blendModeFallback = value !== 'source-over' && this.ctx.globalCompositeOperation !== value;`.

Reply:
I agree with the above. If it is possible to place it in `compatibility.js` I'd highly prefer that, but if that's not possible/makes the code much more difficult, then this is acceptable provided that the above is done, i.e., there is a good comment and it's not bundled unnecessarily.

Reply:
Moved the check to compatibility.js as efficiently as I could. I now override the setter for the property in question, and this check is only done once.